### PR TITLE
feat(handshake): Improve handshake error handling... actually check the response

### PR DESCRIPTION
**Pull request by AndrewWestberg**
We were currently doing a very dumb check on the payload size to determine if a handshake is successful. I've implemented some parsing of the response to validate that a node is in the Allegra era for protocol version and also validate that the network magic value is matching.

### CODE REVIEW

**Comment by mark-stopka** [comment]
@pavlix, can you please review and merge if it's ok from your point of view?

File: src/mux/tcp.rs [hunk 1]
```diff
@@ -26,7 +27,10 @@ pub async fn connect(host: &str, port: u16) -> io::Result<Channel> {
     /* TODO: Consider asynchronous operations */
     let saddr = (host, port).to_socket_addrs()?.nth(0)
         .ok_or(Error::new(ErrorKind::NotFound, "No valid host found!"))?;
-    Ok(Channel::new(TcpStream::connect_timeout(&saddr, Duration::from_secs(2))?))
+    let stream = TcpStream::connect_timeout(&saddr, Duration::from_secs(2))?;
```
Comment (pavlix) [inline src/mux/tcp.rs:30]: This should normally be a separate commit that would get merged immediately.

File: src/protocols/handshake.rs [hunk 2]
```diff
@@ -115,20 +127,72 @@ impl Protocol for HandshakeProtocol {
...
-        if data.len() != 8 {
-            // some payload error
-            let cbor_value: Value = de::from_slice(&data[..]).unwrap();
-            match self.find_error_message(&cbor_value) {
-                Ok(error_message) => {
-                    self.result = Some(Err(error_message));
-                }
-                Err(_) => {
-                    self.result = Some(Err(format!("Unable to parse payload error! {}", hex::encode(data))));
+        let confirm: Value = de::from_slice(&data[..]).unwrap();
```
Comment (pavlix) [inline src/protocols/handshake.rs:130]: We need to first turn this into a separate _output validation_ function. Then the verification function should work as a sequence of checks in the form of _error handling_ rather than _nested matching_.

**Review by pavlix — CHANGES_REQUESTED** [review]
Let's improve the code structure before merging.

File: src/protocols/handshake.rs (hunk 2 above)
Comment (AndrewWestberg) [reply src/protocols/handshake.rs:202]: fixed. Please take a look

**Review by pavlix — APPROVED** [review]
Good to merge. We can make minor changes later.
